pytorch_lightning/callbacks/gradient_accumulation_scheduler.py

```diff
@@ -0,0 +1,56 @@
+import warnings
+
+from .callback import Callback
+
+
+class GradientAccumulationScheduler(Callback):
+    r"""
+    Change gradient accumulation factor according to scheduling.
+
+    Args:
+        scheduling (dict): scheduling in format {epoch: accumulation_factor}
+        .. warning:: Epochs indexing starts from "1" until v0.6.x, but will start from "0" in
+        v0.8.0.
+
+    Example::
+
+        from pytorch_lightning import Trainer
+        from pytorch_lightning.callbacks import GradientAccumulationScheduler
+
+        # at epoch 5 start accumulating every 2 batches
+        accumulator = GradientAccumulationScheduler(scheduling: {5: 2})
+        Trainer(accumulate_grad_batches=accumulator)
+    """
+
+    def __init__(self, scheduling: dict):
```

Comment:
we have not used it before but this typing is a good idea, it may replace the writing types extra in docs
@PyTorchLightning/core-contributors ^^

Reply:
Types are super nice to have as it allows your IDE to quickly spot for common type errors. It's ok to only add types for a subset of your functions in your code and add more with time.

Reply:
sure, could you pls create a "good first issue" to add these types to our codebase... :]

Reply:
I can't add the label: https://github.com/PyTorchLightning/pytorch-lightning/issues/887